test/behaviour/resolution/framework/Resolution.java

```diff
@@ -154,5 +158,32 @@ public CorrectnessException(String message) {
         CompletenessException(String message) {
             super(message);
         }
-    };
+    }
+
+    public static class WrongAnswerSizeException extends RuntimeException {
+        public WrongAnswerSizeException(final int expectedAnswers, final int actualAnswers, final GraqlQuery inferenceQuery) {
+            super(String.format("Query had an incorrect number of answers. Expected %d answers, but found %d " +
+                    "answers, for query :\n %s", expectedAnswers, actualAnswers, inferenceQuery));
+            this.actualAnswers = actualAnswers;
+            this.expectedAnswers = expectedAnswers;
+            this.inferenceQuery = inferenceQuery;
+
+        }
+
+        public int getActualAnswers() {
+            return actualAnswers;
+        }
+
+        public int getExpectedAnswers() {
+            return expectedAnswers;
+        }
+
+        public GraqlQuery getInferenceQuery() {
+            return inferenceQuery;
+        }
+
+        private final int expectedAnswers;
```

Comment:
we tend to put state at the top of the class, then constructors, public methods, last private methods :)

Reply:
Rearranged.